tests/controller.spec.ts

```diff
@@ -166,4 +166,44 @@ describe("controller unit test", () => {
             }
         });
     });
+
+    describe('perfomFullActions', () => {
+        let requireMock: any;
+        before(() => {
```

Comment:
should we use `before**each**` and `after**each**` to increase test case isolation?

Reply:
This only matters for the first unit test

Reply:
We still need good isolation between tests to avoid none deterministic behavior.
For example if we do `.only` for the second test these `before` and `after` blocks would still run
can we guarantee they would not affect the results?